docroot/modules/custom/mass_utility/src/Commands/MassUtilityCommands.php

```diff
@@ -398,4 +399,95 @@ public function queueRevisionsCleanup(array $options = [
 
   }
 
+  /**
+   * Copy D2D nodes into redirect entities.
+   *
+   * @command copyd2d
+   * @option limit
+   * @option min-hits
+   */
+  public function copyD2DRedirects($options = ['limit' => self::REQ, 'min-hits' => 50]) {
+    $query = $this->database->select('node__field_legacy_redirects_legacyurl', 'source')
+      ->fields('source', ['entity_id', 'delta']);
+    $query->addField('source', 'field_legacy_redirects_legacyurl_value', 'source');
+    $query->addField('target', 'field_legacy_redirects_ref_conte_target_id', 'target');
+    if ($options['limit']) {
+      $query->range(0, $options['limit']);
+    }
+    $query->innerJoin('node__field_legacy_redirects_ref_conte', 'target', 'source.entity_id = target.entity_id AND source.delta = target.delta');
+    $records = $query->execute()->fetchAll();
+    foreach ($records as $record) {
+      $record->source = parse_url($record->source, PHP_URL_PATH);
+      $record->target = '/node/' . $record->target;
+      if ($this->isActive($record->source, $options)) {
+        $this->saveRedirect($record);
```

Comment:
I'm getting some duplicate entries on save.  I think we should validate that no redirect already exists for the source path.

Reply:
Along those same lines... should we check to make sure that the path is not already aliased before creating?  Some of the previously legacy prefixes now go to Drupal, and I'm not sure we've always had the validation that a legacy redirect actually points to a legacy prefix, so it's theoretically possible that we could have some overlap here.

Reply:
I'm handling the exceptions that happens with a duplicate save. Do you think we should do more, like try not to save it at all?

Reply:
Actually, nah.  That's fine. It should be relatively limited in scope anyway, I'd hope.